cpp_exercises/day3/phoneBook/tests/tst_phonebook.cpp

```diff
@@ -80,18 +123,51 @@ void testPhoneBook::insertionOperatorTest() {
 
     std::stringstream out;
     out << phoneBook;
-
     const char* resultExpected = R"(1) Calà Gerry
    tel: +39 0123456789
 2) Sordi Alberto
-   tel: +39 111222333
+   tel: +38 111222333
 3) Spencer Bud
    tel: +39 20202020
+4) Carrisi Albano
+   tel: +39 444333444
 )";
-
     QCOMPARE(out.str(), resultExpected);
 }
 
+
+void testPhoneBook::filterTest() {
+    PhoneBook phoneBook;
+    fill(phoneBook);
+
+    {
+        RangeFilter filter(1);
+        std::vector<Contact*> contacts = phoneBook.filter(&filter);
```

Comment:
Questo secondo me è uno di quei casi dove dai test in cui si fa uso dell'API di una classe si nota che c'è qualcosa che stona: se c'è una cosa che non mi aspetto è di dover passare un oggetto filtro sapendo che verrà modificato all'interno.
I filtri solitamente, sia che siano implementati come semplici funzioni sia come oggetti, dovrebbero essere in "sola lettura".